src/v/cluster/rm_stm.cc

```diff
@@ -439,6 +473,22 @@ ss::future<tx_errc> rm_stm::abort_tx(
         _mem_state = mem_state{.term = _insync_term};
     }
 
+    auto origin = get_abort_origin(pid, tx_seq);
+    if (origin == abort_origin::past) {
+        // rejecting a delayed abort command to prevent aborting
+        // a wrong transaction
+        co_return tx_errc::request_rejected;
+    } else if (origin == abort_origin::future) {
+        // impossible situation: before transactional coordinator may issue
+        // abort of the current transaction it should begin it and abort all
+        // previous transactions with the same pid
+        vlog(
+          clusterlog.error,
+          "Rejecting abort because it isn't consistent with "
+          "the current ongoing transaction");
```

Comment:
should we log the id triplets (pid, epoch, txn_seq) expected vs what is. some more context to debug this in a production environment from a bad client connection